File: src/main/java/gargoyle/xenox/game/scr/menu/config/JOptions.java
package gargoyle.xenox.game.scr.menu.config;

import gargoyle.xenox.Xenox;
import gargoyle.xenox.game.sys.JScreen;
import gargoyle.xenox.game.sys.Options;
import gargoyle.xenox.util.i18n.Messages;

import javax.swing.JButton;
import javax.swing.JCheckBox;
import javax.swing.JDialog;
import javax.swing.JPanel;
import javax.swing.SwingUtilities;
import javax.swing.WindowConstants;
import java.awt.BorderLayout;
import java.awt.GridLayout;

public class JOptions extends JDialog {
    public static final String STR_OPTIONS = "options";
    private static final String STR_RESIZE = "resize";
    private static final String STR_SOUND = "sound";
    private final Options options;

    public JOptions(JScreen parent) {
        super(SwingUtilities.getWindowAncestor(parent), parent.getMessages().get(STR_OPTIONS),
                ModalityType.APPLICATION_MODAL);
        options = parent.getOptions();
        setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
        init(parent.getMessages());
    }

    private void init(Messages messages) {
        setLayout(new BorderLayout());
        JPanel pnlOptions = new JPanel(new GridLayout(0, 1));
        JCheckBox chkSound = new JCheckBox(messages.get(STR_SOUND), options.isSound());
        pnlOptions.add(chkSound);
        JCheckBox chkResize = new JCheckBox(messages.get(STR_RESIZE), options.isResize());
        pnlOptions.add(chkResize);
        JPanel pnlButtons = new JPanel(new GridLayout(1, 2));
        JButton btnOk = new JButton(messages.get(Xenox.STR_OK));
        btnOk.addActionListener(e -> {
            options.setSound(chkSound.isSelected());
            options.setResize(chkResize.isSelected());
            setVisible(false);
        });
        pnlButtons.add(btnOk);
        JButton btnCancel = new JButton(messages.get(Xenox.STR_CANCEL));
        btnCancel.addActionListener(e -> {
            chkSound.setSelected(options.isSound());
            chkResize.setSelected(options.isResize());
            setVisible(false);
        });
        pnlButtons.add(btnCancel);
        add(pnlOptions, BorderLayout.CENTER);
        add(pnlButtons, BorderLayout.SOUTH);
    }
}

File: src/main/kotlin/gargoyle/xenox/game/scr/menu/config/JOptions.kt
package gargoyle.xenox.game.scr.menu.config

import gargoyle.xenox.Xenox
import gargoyle.xenox.game.sys.JScreen
import gargoyle.xenox.game.sys.Options
import gargoyle.xenox.util.i18n.Messages
import java.awt.BorderLayout
import java.awt.GridLayout
import java.awt.event.ActionEvent
import javax.swing.JButton
import javax.swing.JCheckBox
import javax.swing.JDialog
import javax.swing.JPanel
import javax.swing.SwingUtilities

class JOptions(parent: JScreen) : JDialog(
    SwingUtilities.getWindowAncestor(parent), parent.messages[STR_OPTIONS],
    ModalityType.APPLICATION_MODAL
) {
    private val options: Options?

    init {
        options = parent.options
        defaultCloseOperation = DO_NOTHING_ON_CLOSE
        init(parent.messages)
    }

    private fun init(messages: Messages?) {
        layout = BorderLayout()
        val pnlOptions = JPanel(GridLayout(0, 1))
        val chkSound = JCheckBox(messages!![STR_SOUND], options!!.isSound)
        pnlOptions.add(chkSound)
        val chkResize = JCheckBox(messages[STR_RESIZE], options.isResize)
        pnlOptions.add(chkResize)
        val pnlButtons = JPanel(GridLayout(1, 2))
        val btnOk = JButton(messages[Xenox.STR_OK])
        btnOk.addActionListener {
            options.isSound = chkSound.isSelected
            options.isResize = chkResize.isSelected
            isVisible = false
        }
        pnlButtons.add(btnOk)
        val btnCancel = JButton(messages[Xenox.STR_CANCEL])
        btnCancel.addActionListener {
            chkSound.isSelected = options.isSound
            chkResize.isSelected = options.isResize
            isVisible = false
        }
        pnlButtons.add(btnCancel)
        add(pnlOptions, BorderLayout.CENTER)
        add(pnlButtons, BorderLayout.SOUTH)
    }

    companion object {
        const val STR_OPTIONS = "options"
        private const val STR_RESIZE = "resize"
        private const val STR_SOUND = "sound"
    }
}
